streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -601,12 +620,14 @@ private void restoreChangelog(final ChangelogMetadata changelogMetadata) {
                 throw new StreamsException("State restore listener failed on restore completed", e);
             }
         }
+
+        return numRecords;
     }
 
     private Map<TopicPartition, Long> committedOffsetForChangelogs(final Set<TopicPartition> partitions) {
         final Map<TopicPartition, Long> committedOffsets;
         try {
-            committedOffsets = fetchCommittedOffsets(partitions, mainConsumer);
+            committedOffsets = fetchCommittedOffsets(partitions, groupId, adminClient);
```

Comment:
Now with the changelog reader accessed by a separate thread, we cannot let it to access the main consumer anymore since the consumer is not thread-safe and we do not allow concurrent access.

So here I'm switching to use the admin client to get committed offsets.